x/ecocredit/server/marketplace/utils.go

```diff
@@ -18,3 +24,184 @@ func isDenomAllowed(ctx sdk.Context, denom string, pk ecocredit.ParamKeeper) boo
 	}
 	return false
 }
+
+type orderOptions struct {
+	autoRetire         bool
+	canPartialFill     bool
+	batchDenom         string
+	retirementLocation string
+}
+
+// fillOrder moves credits according to the order. it will:
+// - update a sell order, removing it if quantity becomes 0 as a result of this purchase.
+// - remove the purchaseQty from the seller's escrowed balance.
+// - add credits to the buyer's tradable/retired address (based on the DisableAutoRetire field).
+// - update the supply accordingly.
+// - send the coins specified in the bid to the seller.
```

Comment:
```suggestion
// fillOrder moves credits and coins according to the order. It will:
// - update a sell order, removing it if quantity becomes 0 as a result of this purchase.
// - remove the purchaseQty from the seller's escrowed balance.
// - add credits to the buyer's tradable/retired address (based on the DisableAutoRetire field).
// - update the supply accordingly.
// - send the coins specified in the bid to the seller.
```